### DISCUSSION TITLE
how to get native call stack and how to get how many times a method is called ?

### DISCUSSION THREAD

**Discussion by birojnayak** [dotnet/dotnet-monitor#2163] [Q&A] [answered]
Hi Guys,
I understand that dotnet-trace can't do native callstack and recommend way is LTTng for Linux, wondering if there is a plan ? Would be happy to contribute if there is a plan and we can enable....

Secondly, how to get JIT related info ? e:g:- how many times a method is being invoked ? can we achieve by using this tool ?

Regards,
Biroj.

**Comment by jander-msft** [comment, marked as answer]
> I understand that dotnet-trace can't do native callstack and recommend way is LTTng for Linux

Could you elaborate on what you are requesting? Are you ask for nettrace (the standard output format of dotnet-trace) to include native stack frames when stack information is gathered for each event? Or are you looking for a general mechanism of capturing stack information at a given point in time. If the latter, we plan to add a `/stack(s)` route to the `dotnet-monitor` tool to collect the stack frames of a target process. These may include native stack frames (I'll have to check the current implementation that is being developed at this time).

> Secondly, how to get JIT related info ? e:g:- how many times a method is being invoked ? can we achieve by using this tool ?

JITing is effectively the translation of managed code to native instructions. These seem to be the events for JIT:
https://docs.microsoft.com/en-us/dotnet/framework/performance/method-etw-events#methodjittingstarted-event
https://docs.microsoft.com/en-us/dotnet/framework/performance/jit-tracing-etw-events

I'm not aware of an event that indicates the number of invocations of any given method. There is the [CLRStackWalk](https://docs.microsoft.com/en-us/dotnet/framework/performance/stack-etw-event) event that is sent that represents the frames of a callstack, but these are only produced when another event is raised.

If there is a broader ask for newer/different functionality, then it's better to log an issue in either https://github.com/dotnet/runtime or https://github.com/dotnet/diagnostics.
